pkg/controller/deployment/util/deployment_util.go

```diff
@@ -816,9 +816,10 @@ func IsRollingUpdate(deployment *extensions.Deployment) bool {
 }
 
 // DeploymentComplete considers a deployment to be complete once its desired replicas equals its
```

Comment:
> We expose a Progressing=True condition with NewReplicaSetAvailable as a reason once a deployment completes.

I meant that we should also mention this in https://kubernetes.io/docs/user-guide/deployments/#complete-deployment (right now it's documented under failed deployment which makes it less obvious)